cfg_support/src/lib.rs

```diff
@@ -0,0 +1,335 @@
+/* When cross compiling, build.rs runs on the host, thus when compiling it
+ * target = host. Use env vars listed in the links below for the actual target.
+ *
+ * https://doc.rust-lang.org/cargo/reference/environment-variables.html
+ * https://doc.rust-lang.org/reference/conditional-compilation.html
+ */
+
+#[derive(Debug)]
+pub struct CfgTarget {
+    pub target_arch: String,
+    pub target_feature: String,
+    pub target_os: String,
+    pub target_family: String,
+    pub target_env: String,
+    pub target_endian: String,
+    pub target_pointer_width: String,
+    pub target_vendor: String,
+}
+
+#[derive(Copy, Clone)]
+pub struct Cfg<'a> {
+    arch: Option<&'a str>,
+    // Not including target_feature here,
+    // it doesn't seem useful for an equality test.
+    //
+    // Usually you'll want to CfgTarget::new().target_feature.contains("....")
+    os: Option<&'a str>,
+    family: Option<&'a str>,
+    env: Option<&'a str>,
+    endian: Option<&'a str>,
+    pointer_width: Option<&'a str>,
+    vendor: Option<&'a str>,
+}
+
+#[derive(Copy, Clone)]
+pub enum CfgBuilder<'a> {
+    Uninitialized,
+    Cond(Cfg<'a>),
+}
+
+fn check_same(target: String, other: Option<&'_ str>) -> bool {
+    Some(target) == other.map(|x: &str| x.to_uppercase())
+}
+fn check_conflict(target: String, other: Option<&'_ str>) -> bool {
+    None == other || check_same(target, other)
+}
+
+impl<'a> Cfg<'a> {
+    pub fn arch_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_arch).into(), self.arch)
+    }
+
+    pub fn os_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_os).into(), self.os)
+    }
+
+    pub fn family_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_family).into(), self.family)
+    }
+
+    pub fn env_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_env).into(), self.env)
+    }
+
+    pub fn endian_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_endian).into(), self.endian)
+    }
+
+    pub fn pointer_width_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_pointer_width).into(), self.pointer_width)
+    }
+
+    pub fn vendor_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_vendor).into(), self.vendor)
+    }
```

Comment:
I wouldn't expect these functions to be used individually, and it would save a lot of lines of code to just include their checks inside the one `check()` function rather than declaring all of these.